### PR TITLE
Add new module graphql-kotlin-spring-server

### PR DESCRIPTION
**Pull request by dkuc84**
Create a new module `graphql-kotlin-spring-server` which is a Spring Boot auto-configuration library for creating GraphQL web application.

### CODE REVIEW

**Comment by smyrick** [comment]
Nice! This is exactly what I was talking about this week. I think it would be good even just for us to move more of this basic spring code to open source since everyone will have to do this same setup

I propose we call it `graphql-kotlin-spring-server` though since Spring Boot is just a library on top of Spring

File: examples/federation/base-app/src/main/kotlin/com/expediagroup/graphql/sample/RoutesConfiguration.kt [hunk 1]
```diff
@@ -16,69 +16,23 @@
...
-    @Bean
-    fun sdlRoute() = router {
-        GET("/sdl") {
-            ok().contentType(MediaType.TEXT_PLAIN).syncBody(schema.print())
-        }
-    }
-
     @Bean
-    fun graphQLToolRoutes() = router {
+    fun playgroundRoute() = coRouter {
```
Comment (smyrick) [inline examples/federation/base-app/src/main/kotlin/com/expediagroup/graphql/sample/RoutesConfiguration.kt]: Do we want to include the playground in the library? Everyone is going to include graphiql or the playground if we don't so we could just add this in.

File: graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/spring/SchemaAutoConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+@ConditionalOnProperty(value = ["graphql.federation.enabled"], havingValue = "false", matchIfMissing = true)
+@Configuration
+class SchemaAutoConfiguration {
+
+    private val logger = LoggerFactory.getLogger(SchemaAutoConfiguration::class.java)
+
+    @Bean
+    @ConditionalOnMissingBean
+    fun schemaConfig(config: GraphQLConfigurationProperties): SchemaGeneratorConfig = SchemaGeneratorConfig(
+            supportedPackages = config.packages
```
Comment (smyrick) [inline graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/spring/SchemaAutoConfiguration.kt]: How do we specify other config options? Do we override this bean or do we need to make the spring config more verbose

File: examples/federation/base-app/src/main/kotlin/com/expediagroup/graphql/sample/RoutesConfiguration.kt (hunk 1 above)
Comment (dkuc84) [reply examples/federation/base-app/src/main/kotlin/com/expediagroup/graphql/sample/RoutesConfiguration.kt]: while certainly useful, `playground` is optional and you don't need it to run a GraphQL server. I removed it from `spring-server` due to the above and also to simplify our logic in how to support custom `/graphql` (and `subscriptions` endpoint) as this value is hard coded in the HTML.

File: graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/spring/SchemaAutoConfiguration.kt (hunk 2 above)
Comment (dkuc84) [reply graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/spring/SchemaAutoConfiguration.kt]: I though the simplest route would be to override the bean. All the auto configuration beans will back off if one provides their own.

**Comment by smyrick** [comment]
Looks good to me, ready for review by others?

File: graphql-kotlin-federation/README.md [hunk 3]
```diff
@@ -163,6 +163,6 @@ type Query {
 
 ## Documentation
 
-There are more examples and documentation in our [Wiki](https://github.com/ExpediaGroup/graphql-kotlin/wiki) or you can view the [javadocs](https://www.javadoc.io/doc/com.expedia/graphql-kotlin) for all published versions.
+There are more examples and documentation in our [Wiki](https://github.com/ExpediaGroup/graphql-kotlin/wiki) or you can view the [javadocs](https://www.javadoc.io/doc/com.expediagroup/graphql-kotlin) for all published versions.
```
Comment (smyrick) [inline graphql-kotlin-federation/README.md]: Can we update the javadocs link to `graphql-kotlin-federation`

**Comment by smyrick** [comment]
@dkuc84 It looks like you need to pull from master and update `DataAndErrors.kt`

**Pull request merged**